### ISSUE TITLE
Dependency symfony/dependency-injection ^3.3 conflicts with symfony/symfony 3.2

### ISSUE THREAD

**Issue by benjaminrau** [FriendsOfSymfony/FOSElasticaBundle#1528]
I expected version ~5.0 to be compatible with symfony/symfony ~3.2 but it requires symfony/dependency-injection ^3.3.

symfony/dependency-injection ^3.3 conflicts with symfony/symfony ~3.2.

Am i missing something?

**Comment by XWB** [comment]
It's because we use `ChildDefinition` that was introduced in `symfony/dependency-injection 3.3`: https://github.com/FriendsOfSymfony/FOSElasticaBundle/commit/a81b59531b9d1d6d5bb1a87233071b2746bb89b2#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780

**Comment by benjaminrau** [comment]
@XWB Thanks for the quick response on an actually unmaintained version of FOSElasticaBundle. Just to be sure if i get it right: So 5.0 is not compatible with Symfony 3.2 any longer? Refering to the README.md that says that 5.0 is compatible with ^3.2|^4

**Comment by XWB** [comment]
As for as I know, different versions of Symfony components can be mixed together. You should be able to install Symfony 3.2 and `symfony/dependency-injection 3.3`.

**Comment by hluup** [comment]
> As for as I know, different versions of Symfony components can be mixed together. You should be able to install Symfony 3.2 and `symfony/dependency-injection 3.3`.

Same issue here. Tried `composer require "symfony/dependency-injection:3.3.*" `
```
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Conclusion: remove symfony/symfony v3.2.14
    - Conclusion: don't install symfony/symfony v3.2.14
    - Conclusion: don't install symfony/symfony v3.2.13
    - Conclusion: don't install symfony/symfony v3.2.12
    - Conclusion: don't install symfony/symfony v3.2.11
    - Conclusion: don't install symfony/symfony v3.2.10
    - Conclusion: don't install symfony/symfony v3.2.9
    - Conclusion: don't install symfony/symfony v3.2.8
    - Conclusion: don't install symfony/symfony v3.2.7
    - Conclusion: don't install symfony/symfony v3.2.6
    - Conclusion: don't install symfony/symfony v3.2.5
    - Conclusion: don't install symfony/symfony v3.2.4
    - Conclusion: don't install symfony/symfony v3.2.3
    - Conclusion: don't install symfony/symfony v3.2.2
    - Conclusion: don't install symfony/symfony v3.2.1
    - Conclusion: don't install symfony/symfony v3.2.0
    - don't install symfony/dependency-injection v3.3.0|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.1|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.10|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.11|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.12|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.13|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.14|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.15|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.16|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.17|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.18|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.2|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.3|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.4|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.5|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.6|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.7|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.8|don't install symfony/symfony v3.2.14
    - don't install symfony/dependency-injection v3.3.9|don't install symfony/symfony v3.2.14
    - Installation request for symfony/symfony 3.2.* -> satisfiable by symfony/symfony[v3.2.0, v3.2.1, v3.2.10, v3.2.11, v3.2.12, v3.2.13, v3.2.14, v3.2.2, v3.2.3, v3.2.4, v3.2.5, v3.2.6, v3.2.7, v3.2.8, v3.2.9].
    - Installation request for symfony/dependency-injection 3.3.* -> satisfiable by symfony/dependency-injection[v3.3.0, v3.3.1, v3.3.10, v3.3.11, v3.3.12, v3.3.13, v3.3.14, v3.3.15, v3.3.16, v3.3.17, v3.3.18, v3.3.2, v3.3.3, v3.3.4, v3.3.5, v3.3.6, v3.3.7, v3.3.8, v3.3.9].
```

**Comment by benjaminrau** [comment]
I tried to provide a solution as PR but i think that as long as there is dependency to DI ^3.3 there is no chance for Sf 3.2 support.

I forked and reverted the corresponding commits which introduced DI ^3.3 in the bundle - the fork works with Sf 3.2:
https://github.com/benjaminrau/FOSElasticaBundle/tree/symfony3.2

I work with that fork as long as we have dependeny to Sf 3.2 in our project. As soon as we update to 3.3 the fork is no longer needed.
